plugin/trino-spooling-filesystem/src/main/java/io/trino/spooling/filesystem/FileSystemSpoolingManager.java

```diff
@@ -258,18 +205,19 @@ private void checkExpiration(FileSystemSpooledSegmentHandle handle)
         }
     }
 
-    private static String base64Encode(Slice slice)
-    {
-        return Base64.getEncoder().encodeToString(slice.getBytes());
-    }
-
-    private static Slice base64Decode(String base64)
+    private TrinoFileSystem fileSystem(FileSystemSpooledSegmentHandle handle)
     {
-        return wrappedBuffer(Base64.getDecoder().decode(base64));
+        if (handle.encryptionKey().isPresent()) {
+            return new EncryptionEnforcingFileSystem(fileSystem, handle.encryptionKey().get());
```

Comment:
I am not a big fan of this one. This is fine in tests but it changes the semantics of interface significantly by putting constraints on `EncryptionKey` passed to `new*File` methods,

it looks it would be trivial to just inline the logic in `openInputStream`, `createOutputStream` and `directLocation` and not use `EncryptionEnforcingFileSystem` at all

Reply:
I actually did it before but changed that in the meantime.